on_node_start.sh

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env bash
+
+# https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425#file-bash_strict_mode-md
+set -euo pipefail
+
+. /etc/parallelcluster/cfnconfig
+
+echo "Node type: ${cfn_node_type}"
+
+function configure_yum() {
+    cat >>/etc/yum.conf <<EOF
+assumeyes=1
+clean_requirements_on_remove=1
+EOF
+}
+
+function install_fuse_overlayfs() {
+    yum -q install buildah
+
+    if [[ ! -e /dev/fuse ]]; then
+        mknod /dev/fuse -m 0666 c 10 229
+    fi
+
+    pushd "$(mktemp -d)"
+    git clone --depth 1 https://github.com/containers/fuse-overlayfs.git ./overlay
+    pushd ./overlay
+    buildah bud -v "$PWD:/build/fuse-overlayfs" -t fuse-overlayfs -f ./Containerfile.static.ubuntu .
+    cp fuse-overlayfs /usr/bin/
+
+}
+
+function yum_cleanup() {
+    yum -q clean all
+    rm -rf /var/cache/yum
+}
+function install_podman() {
+    yum -q install fuse-overlayfs slirp4netns podman
+}
+
+function install_head_node_dependencies() {
+    # MariaDB repository setup
+    curl -sS https://downloads.mariadb.com/MariaDB/mariadb_repo_setup | bash
+
+    yum_cleanup
+
+    yum -q update
+
+    yum -q install epel-release
+    yum-config-manager -y --enable epel
+
+    # mariadb
+    yum -q install MariaDB-server
+
+    # podman
+    install_podman
+
+    # Install go-task, see https://taskfile.dev/install.sh
+    sh -c "$(curl --location https://taskfile.dev/install.sh)" -- -d -b /usr/local/bin
+
+    yum_cleanup
+}
+
+function create_and_save_slurmdb_password() {
+    local slurm_password_file="/root/slurmdb.password"
+    if [[ -e "$slurm_password_file" ]]; then
+        echo "Error: create_and_save_slurmdb_password() was called when a password file already exists" >&2
+        return 1
+    fi
+
+    echo -n "$(openssl rand -hex 32)" >"$slurm_password_file"
+}
+
+function configure_slurm_database() {
+
+    systemctl enable mariadb.service
+    systemctl start mariadb.service
+
+    create_and_save_slurmdb_password
+
+    local slurmdbd_password
+    slurmdbd_password="$(cat /root/slurmdb.password)"
+
+    local slurmdbd_user="slurm"
+    local slurmdb_name="slurm_acct_db"
+
+    mysql --wait -e "CREATE USER '${slurmdbd_user}'@'localhost' identified by '${slurmdbd_password}'"
+    mysql --wait -e "CREATE DATABASE ${slurmdb_name}"
+    mysql --wait -e "GRANT ALL ON ${slurmdb_name}.* to '${slurmdbd_user}'@'localhost' identified by '${slurmdbd_password}'"
```

Comment:
This would fail is `slurmdb_name` has spaces or other special characters in but I presume that doesn't really matter here as the variable is locally defined within this script and not depending on any user configuration or input?